components/org.wso2.carbon.status.dashboard.core/src/main/java/org/wso2/carbon/status/dashboard/core/dbhandler/StatusDashboardWorkerDBHandler.java

```diff
@@ -95,79 +95,68 @@ public StatusDashboardWorkerDBHandler() {
             creteDetailsDB();
         } else {
             throw new RDBMSTableException(DATASOURCE_ID + " Could not find. Hence cannot initialize the status " +
-                    "dashboard.");
+                    "dashboard. Please check database is available");
         }
     }
 
     // TODO: 11/2/17 improve for all databases
     private void creteConfigurationDB() {
         Connection conn = this.getConnection();
         String resolved = tableCheckQuery.replace(PLACEHOLDER_TABLE_NAME, WORKER_CONFIG_TABLE);
-        PreparedStatement tableCheckstmt = null;
-        try {
-            tableCheckstmt = conn.prepareStatement(resolved);
-            if (!DBHandler.getInstance().isTableExist(tableCheckstmt)) {
-                if (!isConfigTableCreated) {
-                    String resolvedTableCreateQuery = "CREATE TABLE IF NOT EXISTS WORKERS_CONFIGURATION (\n" +
-                            "WORKERID VARCHAR(255) PRIMARY KEY,\n" +
-                            "HOST VARCHAR(500),\n" +
-                            "PORT INT\n" +
-                            ");";
-                    try {
-                        PreparedStatement stmt = conn.prepareStatement(resolvedTableCreateQuery);
-                        stmt.execute();
-                        isConfigTableCreated = true;
-                        stmt.close();
-                    } catch (SQLException e) {
-                        logger.error("Error creating table please create manually ." + WORKER_CONFIG_TABLE, e);
-                    }
+        if (!DBHandler.getInstance().isTableExist(conn,resolved)) {
+            if (!isConfigTableCreated) {
```

Comment:
Hardcoded SQL query?

Reply:
Fixed in 3748538e04efaf2535e6b6c74f7b4024e27fd6d9